src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/managehearings/ManageHearingActionService.java

```diff
@@ -115,6 +118,47 @@ public void performAddHearing(FinremCaseDetails finremCaseDetails, String authTo
         hearingWrapper.setWorkingHearing(null);
     }
 
+    /**
+     * Vacates a selected hearing from the case, transferring its details to the vacated hearings collection.
+     * Removes the specified hearing from the active hearings list and creates a corresponding vacated hearing entry
+     * with additional information provided by the user.
+     *
+     * @param finremCaseDetails case details containing hearing and case data
+     */
+    public void performVacateHearing(FinremCaseDetails finremCaseDetails) {
+        FinremCaseData caseData = finremCaseDetails.getData();
+        ManageHearingsWrapper hearingsWrapper = caseData.getManageHearingsWrapper();
+
+        VacateHearingAction vacateHearingInput = hearingsWrapper.getVacateHearingSelection();
+        UUID selectedHearingId = UUID.fromString(vacateHearingInput.getChooseHearings().getValue().getCode());
+
+        List<ManageHearingsCollectionItem> hearings = new ArrayList<>(hearingsWrapper.getHearings());
+        List<ManageVacatedHearingsCollectionItem> vacateHearings = new ArrayList<>(
+            ofNullable(hearingsWrapper.getVacatedHearingsCollection()).orElseGet(ArrayList::new)
+        );
+
+        ManageHearingsCollectionItem hearingToVacate = hearings.stream()
+            .filter(item -> selectedHearingId.equals(item.getId()))
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("No hearing found with ID: " + selectedHearingId));
+
+        hearings.remove(hearingToVacate);
+
+        VacateOrAdjournedHearing vacatedHearing = VacateOrAdjournedHearing.fromHearingToVacatedHearing(hearingToVacate, vacateHearingInput);
+
+        ManageVacatedHearingsCollectionItem vacatedItem = ManageVacatedHearingsCollectionItem.builder()
+            .id(hearingToVacate.getId())
+            .value(vacatedHearing)
+            .build();
+
+        vacateHearings.add(vacatedItem);
+
+        hearingsWrapper.setHearings(hearings);
+        hearingsWrapper.setVacatedHearingsCollection(vacateHearings);
```

Comment:
You've been manipulating these list, after they were assigned to pointers with getHearings and getVacatedHearingsCollection.

So you don't need lines 156 and 157.

You can compare the memory addresses for hearings and vacateHearings and they mention hearingsWrapper.getHearings() and hearingsWrapper.setVacatedHearingsCollection() respectively

(you can check that in the debugger)